projects/components/src/action-menu/action-menu.component.spec.ts

```diff
@@ -0,0 +1,373 @@
+                    paused: false,
+                },
+            ];
+            const action = {
+                textKey: 'action',
+                handler: (rec: any[], data: Blah) => {
+                    return Promise.resolve(JSON.stringify(data));
+                },
+                handlerData,
+                availability: () => true,
+            };
+            testHostComponent.actionMenuComp.selectedEntities = selectedEntities;
+            const spy = spyOn(action, 'handler').and.returnValue(null);
+            testHostComponent.actionMenuComp.runActionHandler(action);
+            expect(spy).toHaveBeenCalledWith(selectedEntities, handlerData);
+        });
+    });
+    describe('isActionDisabled', () => {
+        it('returns the output of disabled, if its a function or just disabled if its a boolean', () => {
+            const action = {
+                textKey: 'action',
+                handler: () => {},
+                disabled: () => true,
+            };
+            expect(testHostComponent.actionMenuComp.isActionDisabled(action)).toBeTruthy();
+            // @ts-ignore
+            action.disabled = false;
+            expect(testHostComponent.actionMenuComp.isActionDisabled(action)).toBeFalsy();
+        });
+    });
+    describe('shouldShowContextualActions', () => {
+        it('returns true only if there are selected entities', () => {
+            expect(testHostComponent.actionMenuComp.shouldShowContextualActions).toBeFalsy();
+            testHostComponent.actionMenuComp.selectedEntities = [
+                {
+                    value: 'foo',
+                    paused: false,
+                },
+            ];
+            expect(testHostComponent.actionMenuComp.shouldShowContextualActions).toBeTruthy();
+        });
+    });
+    describe('getFlattenedActionList', () => {
+        it('returns nested featured actions by adding them to a flattened list', () => {
+            const flatList = (testHostComponent.actionMenuComp as any).getFlattenedActionList(
+                NESTED_ACTIONS,
+                ActionType.CONTEXTUAL_FEATURED
+            );
+            expect(flatList.length).toEqual(4);
+            flatList.forEach(action => {
+                expect(action.actionType).toEqual(ActionType.CONTEXTUAL_FEATURED);
+            });
+        });
+    });
+});
+
+interface Record {
+    value: string;
+    paused: boolean;
+}
+
+interface Blah {
+    foo: string;
+    bar: string;
+}
+
+type HandlerData = Record[] | Blah;
+
+@Component({
+    template: `
+        <vcd-action-menu [actions]="actions" [actionDisplayConfig]="actionDisplayConfig"> </vcd-action-menu>
+    `,
+})
+class TestHostComponent<R extends Record> {
+    @ViewChild(ActionMenuComponent, { static: true }) actionMenuComp: ActionMenuComponent<R, HandlerData>;
+
+    actionDisplayConfig: ActionDisplayConfig;
+
+    actions: ActionItem<R, HandlerData>[] = [];
+}
+
+const ACTION_DISPLAY_CONFIG = {
```

Comment:
Do not re-use objects , create a function that returns a new object so that every caller gets a separate instance. Change all instances in change